contrib/babelfishpg_tsql/src/databasepropertyex.c

```diff
@@ -18,6 +18,23 @@
 #include "catalog.h"
 #include "catalog.h"
 
+static char*
+get_collation_name(Datum dbname)
+{
+	HeapTuple	tuple;
+	Form_sysdatabases sysdb;
+
+	tuple = SearchSysCache1(SYSDATABASENAME, dbname);
+
+	if (!HeapTupleIsValid(tuple))
+		return NULL;
+
+	sysdb = ((Form_sysdatabases) GETSTRUCT(tuple));
+
+	ReleaseSysCache(tuple);
+	return sysdb->default_collation.data;
+}
```

Comment:
should we return a copy here ? instead of directly passing the cache entry ?

Reply:
Yes, updated